nucliadb/nucliadb/search/search/chat/query.py

```diff
@@ -79,59 +150,57 @@ async def chat(
 
     ident, generator = await predict.chat_query(kbid, chat_model)
 
-    async def generate_answer(
-        results: KnowledgeboxFindResults,
-        kbid: str,
-        predict: PredictEngine,
-        generator: AsyncIterator[bytes],
-        features: List[ChatOptions],
-    ):
-        if ChatOptions.PARAGRAPHS in features:
-            bytes_results = base64.b64encode(results.json().encode())
-            yield len(bytes_results).to_bytes(length=4, byteorder="big", signed=False)
-            yield bytes_results
-
-        answer = []
-        async for data in generator:
-            answer.append(data)
-            yield data
-
-        if ChatOptions.RELATIONS in features:
-            yield END_OF_STREAM
-
-            text_answer = b"".join(answer)
-
-            detected_entities = await predict.detect_entities(
-                kbid, text_answer.decode()
-            )
-            relation_request = RelationSearchRequest()
-            relation_request.subgraph.entry_points.extend(detected_entities)
-            relation_request.subgraph.depth = 1
-
-            relations_results: List[RelationSearchResponse]
-            (
-                relations_results,
-                _,
-                _,
-                _,
-            ) = await node_query(
-                kbid, Method.RELATIONS, relation_request, chat_request.shards
-            )
-            yield base64.b64encode(
-                (
-                    await merge_relations_results(
-                        relations_results, relation_request.subgraph
-                    )
-                )
-                .json()
-                .encode()
-            )
-
     return StreamingResponse(
-        generate_answer(find_results, kbid, predict, generator, chat_request.features),
+        generate_answer(
+            find_results,
+            kbid,
+            user_id,
+            client_type,
+            origin,
+            predict,
+            generator,
+            chat_request=chat_request,
+        ),
         media_type="plain/text",
         headers={
             "NUCLIA-LEARNING-ID": ident or "unknown",
             "Access-Control-Expose-Headers": "NUCLIA-LEARNING-ID",
         },
     )
+
+
+def audited_generator(
```

Comment:
hmmm, I'm not sure the wrapper approach is more readable than just doing the check and audit directly in the code like before

Reply:
actually you are right, this is too complicated